drivers/serial.c

```diff
@@ -37,14 +37,45 @@ static struct {
     unsigned init;
 } input_state;
 
-extern io_port_t com_ports[2];
+io_port_t __data_rmode com_ports[4];
+
+static const char *com_port_names[] = {
+    [COM1_PORT] = "COM1",
+    [COM2_PORT] = "COM2",
+    [COM3_PORT] = "COM3",
+    [COM4_PORT] = "COM4",
+};
```

Comment:
Should use `COM1...COM4` as indices instead to avoid creating a huge, almost empty array.

Reply:
Very good point, thanks!